lib/Sema/CSSimplify.cpp

```diff
@@ -10014,6 +10013,12 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
         impact = 10;
     }
 
+    if (locator->isLastElement<LocatorPathElt::UnresolvedMemberChainResult>()) {
+      // If this is a contextual failure for an unresolved member, then increase
```

Comment:
@Jumhyn Just out of curiosity, in which situation this is necessary? I was recently looking into some situations involving ambiguity so I'm just curious about this in the particular case o member chain and what kind of ambiguity is produced by it

Reply:
I have to verify whether this is still necessary since the implementation has been reworked since this was added. I can't remember the specific diagnostic that this was meant to deal with, but if I find that it's still necessary I'll circle back with more info!

Reply:
This would be great to clarify!

Reply:
Good and bad news—the good news is that it looks like the broader changes to our approach have removed the need for this change; the bad news is that I am unable to shed more light on why it was necessary in the first place. Sorry @LucianoPAlmeida!

Reply:
Ah, I missed that one, no worries @Jumhyn, I was just curious thanks for looking it up :)